Update msrv deps

benalleng opened the pull request:
I took the liberty to update the versions on many of the dependencies that don't necessarily need to be updated based solely on our new msrv.

Every existing dep should now be on an msrv of 1.85.0.  Our ohttp-relay is now using the new v0.0.11 release with these changes as well.

There were a few api changes that were made in the source but most of those changes are minor and can be seen in separate commits

spacebear21 commented:
We will also want to update the README to remove the [pinned deps instructions](https://github.com/payjoin/rust-payjoin/blob/master/README.md#payjoin-1).

benalleng commented:
I have currently pointed `ohttp-relay` to my own draft branch https://github.com/benalleng/rust-payjoin/blob/386a20591209d1d67f57da2981d01baa80cd287c/payjoin-test-utils/Cargo.toml#L18 but hopefully we can get this PR merged and then continue to move everything forward

benalleng commented:
@DanGould would you like this PR to similarly split into seperate commits like in the ohttp relay https://github.com/payjoin/ohttp-relay/pull/66?

DanGould commented:
It doesn't need to be incredibly granular but it would be nice if the source code changes could be associated with specific commits for just the relevant dependency upgrades

benalleng commented:
Note that we are now pointing directly to the main branch on ohttp-relay instead of a release candidate

DanGould reviewed (approved):
I'll approve this because it's not a regression, but one status code handler was changed from a wrong implementation to another wrong implementation that I'd like to see addressed in an immediate follow up. I noticed that the defined `anyhow` version also varied. Seems OK to me.

ACK 083620dddc67be527918d1d5a0e077f328597b95

benalleng commented on payjoin-directory/Cargo.toml:
```diff
@@ -18,23 +18,23 @@ resolver = "2"
 [dependencies]
-anyhow = "1.0.71"
-bitcoin = { version = "0.32.4", features = ["base64", "rand-std"] }
+anyhow = "1.0.89"
```
This is a mistake! i will change it

benalleng commented:
Ok, the anyhow error was a mistake I made when splitting these commits up but is now fixed, as well I added a description for the deprecation of get_async_connection for `redis::Client` in https://github.com/payjoin/rust-payjoin/pull/959/commits/5c44c022739b2e2dc05424627b215855afa3938c

DanGould reviewed (approved):
Legend

ACK 7eaa50e7505709fbdcb442710ef2dd80360a6567
